src/types/options/index.ts

```diff
@@ -1,31 +1,51 @@
 import { Option } from '@slack/types';
 import { StringIndexed, XOR } from '../helpers';
 import { AckFn } from '../utilities';
+import { ViewOutput } from '../view/index';
 
 /**
  * Arguments which listeners and middleware receive to process an options request from Slack
  */
 export interface SlackOptionsMiddlewareArgs<Source extends OptionsSource = OptionsSource> {
-  payload: OptionsRequest<Source>;
+  payload: OptionsPayloadFromType<Source>;
   body: this['payload'];
   options: this['payload'];
   ack: OptionsAckFn<Source>;
 }
 
 /**
- * A request for options for a select menu with an external data source, wrapped in the standard metadata. The menu
- * can have a source of Slack's Block Kit external select elements, dialogs, or legacy interactive components.
- *
- * This describes the entire JSON-encoded body of a request.
+ * All sources from which Slack sends options requests.
  */
-export interface OptionsRequest<Source extends OptionsSource = OptionsSource> extends StringIndexed {
+export type OptionsSource = 'interactive_message' | 'dialog_suggestion' | 'block_suggestion';
+
+export type SlackOptions = BlockSuggestion | InteractiveMessageSuggestion | DialogSuggestion;
```

Comment:
This may sound a bit unnatural but this is consistent with other union types (e.g., SlackAction, SlackEvent, SlackShortcut, SlackViewAction)